libs/langchain-community/src/chat_models/watsonx_ai.ts

```diff
@@ -0,0 +1,157 @@
+import { CallbackManagerForLLMRun } from "@langchain/core/callbacks/manager";
+import {
+  type BaseChatModelParams,
+  SimpleChatModel,
+} from "@langchain/core/language_models/chat_models";
+import {
+  AIMessageChunk,
+  BaseMessage,
+  ChatMessage,
+} from "@langchain/core/messages";
+import { ChatGenerationChunk } from "@langchain/core/outputs";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+import type {
+  WatsonModelParameters,
+  WatsonxAIParams,
+} from "../types/watsonx-types.js";
+import { WatsonApiClient } from "../utils/watsonx-client.js";
+
+export class WatsonxAIChat extends SimpleChatModel {
+  private readonly watsonApiClient: WatsonApiClient;
+
+  readonly modelId!: string;
+
+  readonly modelParameters?: WatsonModelParameters;
+
+  readonly projectId!: string;
+
+  constructor(fields: WatsonxAIParams & BaseChatModelParams) {
+    super(fields);
+
+    const {
+      clientConfig = {},
+      modelId = "meta-llama/llama-2-70b-chat",
+      modelParameters,
+      projectId = getEnvironmentVariable("WATSONX_PROJECT_ID") ?? "",
+    } = fields;
+
+    this.modelId = modelId;
+    this.modelParameters = modelParameters;
+    this.projectId = projectId;
+
+    const {
+      apiKey = getEnvironmentVariable("IBM_CLOUD_API_KEY"),
+      apiVersion = "2023-05-29",
+      region = "us-south",
+    } = clientConfig;
+
+    if (!apiKey) {
+      throw new Error("Missing IBM Cloud API Key");
+    }
+
+    if (!this.projectId) {
+      throw new Error("Missing WatsonX AI Project ID");
+    }
+
+    this.watsonApiClient = new WatsonApiClient({
+      apiKey,
+      apiVersion,
+      region,
+    });
+  }
+
+  protected _formatMessagesAsPrompt(messages: BaseMessage[]): string {
```

Comment:
We should make a `Llama` adapter or something...

Reply:
Yes definitely, I was thinking about it too. Right now this is blatantly copy pasted from the Ollama model and assumes the user only uses llama based chat model. However Watson offers the ability to run many models, including spinning any model from HF...